Initial project commit

sylviamclaughlin opened the pull request:
# Summary | Résumé

Initial project implementation of the AWS access portal. Right now it does not do anything but the CSS and basic layout is initially done and it looks something like this:

![Screenshot 2024-04-10 at 4 30 23 PM](https://github.com/cds-snc/aws-access-portal/assets/85905333/c94729f4-d662-4c6a-8d1e-3fd5e138c237)

sylviamclaughlin committed:
Initial project structure

sylviamclaughlin committed:
Removing a # statement

sylviamclaughlin committed:
Deleting unecesary files

sylviamclaughlin committed:
Adding frontend components

sylviamclaughlin committed:
Adding url rule and removing unecessary lines

sylviamclaughlin committed:
Removing .gitignnore file plus adding more to original .gitignore

sylviamclaughlin committed:
Restoring original .gitignore file and removing the frontend one

sylviamclaughlin committed:
Committing new requirements

patheard commented on aws_access_portal/aws_access_portal/settings.py:
```diff
@@ -0,0 +1,128 @@
+# Internationalization
+# https://docs.djangoproject.com/en/5.0/topics/i18n/
+
+LANGUAGE_CODE = "en-us"
```
Should this be `en-ca`?

sylviamclaughlin replied:
Yes, it should be - thank you for noticing it. I have changed it.

patheard reviewed (commented):
Looks good!  Had a few comments/questions about some of the files:

- Do we need the `frontend/src/components/icons/*.vue` files?  They all looked Vue specific (but I also have zero experience with Vue).
- Do we need to include any of the `*.css` files if we're using Design System?  I didn't see them getting included in any template files but also could have missed the includes.
- The `favicon.ico`, `sig-blk-en.svg` and `wmms-blk.svg` are being included twice (static folder and also in assets folder).

sylviamclaughlin commented:
> Looks good! Had a few comments/questions about some of the files:
> 
> * Do we need the `frontend/src/components/icons/*.vue` files?  They all looked Vue specific (but I also have zero experience with Vue).
> * Do we need to include any of the `*.css` files if we're using Design System?  I didn't see them getting included in any template files but also could have missed the includes.
> * The `favicon.ico`, `sig-blk-en.svg` and `wmms-blk.svg` are being included twice (static folder and also in assets folder).

Thanks for the feedback Pat! Here are the answers to your questions:
1. We did not specifically need them at the moment but I thought that I should still keep them in there since I was planning to use them. At the same time though, you are totally right that they should be removed since I am not using them now and they are cluttering everything. I have removed.
2. I included the design system's CDN by having this in frontend/index.html:
```
 <!-- GC Design System -->
    <link rel="stylesheet" href="https://cdn.design-system.alpha.canada.ca/@cdssnc/gcds-components@latest/dist/gcds/gcds.css">
    <script type="module" src="https://cdn.design-system.alpha.canada.ca/@cdssnc/gcds-components@latest/dist/gcds/gcds.esm.js"></script>
    <script nomodule src="https://cdn.design-system.alpha.canada.ca/@cdssnc/gcds-components@latest/dist/gcds/gcds.js"></script>
```
But you do bring up a good point and I just checked again installation instructions and saw that you can natively install it for Vue. I will make a PR right after this one to take care of it (after testing).
3. I removed the images in the Django backend since you are right that they are not used there - they are just used in the frontend.

sylviamclaughlin committed:
Removing icons that are not needed and images in Django since they are already contained in the Vue frontend

sylviamclaughlin committed:
Changing the language to en-ca
